modules/renter/renter.go

```diff
@@ -338,6 +339,39 @@ func (r *Renter) ProcessConsensusChange(cc modules.ConsensusChange) {
 	r.mu.Unlock(id)
 }
 
+// validateSiapath checks that a Siapath is a legal filename.
+// ../ is disallowed to prevent directory traversal, and paths must not begin
+// with / or be empty.
+func validateSiapath(siapath string) error {
+	if siapath == "" {
+		return ErrEmptyFilename
+	}
+	if siapath == ".." {
+		return errors.New("siapath cannot be ..")
+	}
+	if siapath == "." {
+		return errors.New("siapath cannot be .")
+	}
+	// check prefix
+	if strings.HasPrefix(siapath, "/") {
+		return errors.New("siapath cannot begin with /")
+	}
+	if strings.HasPrefix(siapath, "../") {
+		return errors.New("siapath cannot begin with ../")
+	}
+	if strings.HasPrefix(siapath, "./") {
+		return errors.New("siapath connot begin with ./")
+	}
+	// check Contains
+	if strings.Contains(siapath, "/../") {
+		return errors.New("siapath cannot contain /../")
+	}
+	if strings.Contains(siapath, "/./") {
+		return errors.New("siapath cannot contain /./")
+	}
+	return nil
```

Comment:
I think these are all variants of:
```go
for _, pathElem := range strings.Split(siapath, "/") {
    if pathElem == "." || pathElem == ".." {
        return errors.New("siapath cannot contain . or .. elements")
    }
}
```

Reply:
Would you prefer your method?  I find mine easier to understand albeit slightly longer.

Reply:
hmm. I do prefer the loop actually. It's more clear to me that we're explicitly searching for standalone instances of '.' or '..', instead of doing an enumeration which risks missing a particular corner case.